Configuration/NodeTypes.Overrides.yaml

```diff
@@ -0,0 +1,9 @@
+'Neos.Neos:Node':
+  options:
+    fusion:
+      prototypeGenerator: ~
```

Comment:
By making them configurable we took a step forward in terms of flexibility, I'd like to see that being used to achieve ease of integration without getting in the way..

Fortunately the properties will only be loaded if used, so there isn't a performance penalty added them to ease developer flow.

The point about developer being beginner friendly can be interpreted both ways, it can ease integration by not having to add all properties manually in Fusion to have access to them in the template, however it's still magic that needs explaining.

Can you elaborate why you find it difficult to explain while teaching Neos? To me it seems fairly straight forward that for every document and content node type, a template fusion object with a default template path and all properties mapped is generated automatically and you can override/extend those as needed.

Now the problem is that the document one often gets in the way, however if the best practice becomes using `.Document` then the generator should generate that instead, avoiding having to do the `prototype(Acme.Demo:Page) >`.